File: shell/dnf_release.go
package main

import (
	"fmt"
	"log"
	"os"
	"os/exec"
	"strings"
)

func main() {
	var dnfRelease *exec.Cmd = exec.Command(
		"sudo",
		"dnf",
		"system-upgrade",
		"download",
		"--releasever=41",
	)

	var dnfReleaseOutput strings.Builder

	dnfRelease.Stdout = &dnfReleaseOutput

	var dnfReleaseError error = dnfRelease.Run()

	if dnfReleaseError != nil {
		log.Fatalln("DNF Release Download Failed: \n", dnfReleaseError)
	} else {
		fmt.Println("Executed Command: \n", dnfRelease)
		fmt.Println("Running DNF Release Download: \n", dnfReleaseOutput.String())

		os.Exit(0)
	}
}

File: upd/src/upgrades/dnf_release.rs
use std::{
    io::Error,
    process::{Command, ExitCode, Output},
    result::{
        Result,
        Result::{Err, Ok},
    },
    string::String,
};

// DNF System Release Upgrade
pub fn release_dnf() -> ExitCode {
    let dnf_release: Result<Output, Error> = Command::new("dnf")
        .arg("system-upgrade")
        .arg("download")
        .arg("--releasever=41")
        .output();

    match dnf_release {
        Ok(release) => {
            println!("Command Output: {:#?}", String::from_utf8(release.stdout));
            println!("Status: {}", release.status);
        }
        Err(error) => {
            eprintln!("System Release Upgrade Error for DNF {}", error);
            return ExitCode::FAILURE;
        }
    };

    return ExitCode::SUCCESS;
}
